qutebrowser/browser/greasemonkey.py

```diff
@@ -119,9 +119,9 @@ def code(self):
         template = jinja.js_environment.get_template('greasemonkey_wrapper.js')
         return template.render(
             scriptName=javascript.string_escape(
-                "/".join([self.namespace or '', self.name])),
+                "/".join([self.namespace or '', self.name or ''])),
```

Comment:
@toofar In your commit message, you say "While other things may break if a script doesn't have a name" - so allowing an empty name sounds like more trouble than it's worth? Why would we allow that?

Reply:
Okay, that's fair. I have tried to make it fail early if there is no name (ValueError from __init__) but also provide `filename` value to fall back to if there is no name key in the metadata. Now there is some fallback logic in `__init__` and some in the `parse()` factory method. Maybe the includes fallback should be move up there too but too, but have a script explicitly have just one include line with `*` as the value might be something a caller wants to know. I'm on the fence.

I still left the one place that uses `script_meta` with the fallback because I like the idea of just being able to drop some arbitrary JS in a folder and have it run everywhere (that JS is enabled) and it doesn't cost us anything.